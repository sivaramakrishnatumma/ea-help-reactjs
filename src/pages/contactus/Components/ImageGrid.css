.images-container {
  padding-left: 100%;
  padding-right: 100%;
  padding-top: 20px;
  padding-bottom: 30px;
  margin-left: -100%;
  margin-right: -100%;
  margin-top: 27px;
  background-color: rgb(243, 243, 243);
  overflow-x: hidden;
}

.images-grid {
  display: flex;
  flex-wrap: wrap;
  margin-bottom: 20px;
}

.product-item {
  width: calc((100% / 6) - 20px);
  min-width: 200px;
  margin: 10px;
  cursor: pointer;
}

.product-item img {
  background: #fff;
}