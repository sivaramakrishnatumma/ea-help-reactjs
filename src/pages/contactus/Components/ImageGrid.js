import React from "react";

import Image from "../../../shared/components/UIElements/Image/Image";
import "./ImageGrid.css";

const ImageGrid = props => {
  const { items, onSelect } = props;
  return (
    <div className="images-container">
      <div className="images-grid">
        {items &&
          items.map(item => {
            return (
              <div
                className="product-item"
                key={item.id}
                title={item.name}
                onClick={() => onSelect(item)}
              >
                <Image src={item.imageUrl} alt={item.title}></Image>
              </div>
            );
          })}
      </div>
    </div>
  );
};

export default ImageGrid;
