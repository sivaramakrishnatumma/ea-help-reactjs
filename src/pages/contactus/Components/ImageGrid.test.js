import React from "react";
import { mount } from "enzyme";
import ImageGrid from "./ImageGrid";

describe("ImageGrid component test case", () => {
  it("should test ImageGrid component with default props", () => {
    const props = {
      items: [
        {
          id: "test",
          name: "test-name",
          imageUrl: "test-url"
        }
      ],
      onSelect: jest.fn()
    };
    const wrapper = mount(<ImageGrid {...props} />);
    wrapper.find(".product-item").simulate("click");
    expect(props.onSelect).toHaveBeenCalled();
    expect(wrapper).toMatchSnapshot();
  });
});
