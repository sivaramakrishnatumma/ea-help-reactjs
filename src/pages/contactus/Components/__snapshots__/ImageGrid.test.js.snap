// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageGrid component test case should test ImageGrid component with default props 1`] = `
<ImageGrid
  items={
    Array [
      Object {
        "id": "test",
        "imageUrl": "test-url",
        "name": "test-name",
      },
    ]
  }
  onSelect={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "id": "test",
            "imageUrl": "test-url",
            "name": "test-name",
          },
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <div
    className="images-container"
  >
    <div
      className="images-grid"
    >
      <div
        className="product-item"
        key="test"
        onClick={[Function]}
        title="test-name"
      >
        <Image
          alt="test-name"
          src="test-url"
        >
          <img
            className="img-fluid undefined"
            src="test-url"
          />
        </Image>
      </div>
    </div>
  </div>
</ImageGrid>
`;
